File: app/src/main/java/com/mumanit/foursquareclient/data/cache/VenuesCache.java
package com.mumanit.foursquareclient.data.cache;

import com.mumanit.foursquareclient.domain.model.VenueData;
import java.util.List;

import rx.Observable;

/**
 * Created by pmuciek on 7/9/17.
 */

public interface VenuesCache {
    void save(List<VenueData> venueDataList);
    //Observable<Boolean> save(List<VenueData> venueDataList);
    Observable<List<VenueData>> loadVenues();
}

File: app/src/main/java/com/mumanit/foursquareclient/data/cache/VenuesCache.kt
package com.mumanit.foursquareclient.data.cache

import com.mumanit.foursquareclient.domain.model.VenueDomainModel

import rx.Observable

interface VenuesCache {
    fun save(venueDataList: List<VenueDomainModel>)
    //Observable<Boolean> save(List<VenueDomainModel> venueDataList);
    fun loadVenues(): Observable<List<VenueDomainModel>>
    suspend fun loadVenuesSuspend() : List<VenueDomainModel>
}
